src/main/java/carpet/settings/CarpetSettings.java

```diff
@@ -89,9 +83,53 @@
     @Rule( desc = "Explosions won't destroy blocks", category = CREATIVE )
     public static boolean explosionNoBlockDamage = false;
 
-    @Rule( desc = "Removes random TNT momentum when primed", category = CREATIVE )
+    @Rule( desc = "Removes random TNT momentum when primed", category = {CREATIVE, TNT} )
     public static boolean tntPrimerMomentumRemoved = false;
 
+    @Rule( desc = "TNT causes less lag when exploding in the same spot and in liquids", category = TNT)
+    public static boolean optimizedTNT = false;
+
+    @Rule( desc = "Sets the tnt random explosion range to a fixed value", category = TNT, options = "-1",
+            validate = TNTRandomRangeValidator.class, extra = "Set to -1 for default behavior")
+    public static double tntRandomRange = -1;
+
+    private static class TNTRandomRangeValidator extends Validator<Double> {
+        @Override
+        public Double validate(ServerCommandSource source, ParsedRule<Double> currentRule, Double newValue, String string) {
+            if (newValue == -1)
+                return newValue;
+            if (!optimizedTNT)
+                return null;
+            return newValue >= 0 ? newValue : null;
+        }
+
+        @Override
+        public String description() {
+            if (!optimizedTNT)
+                return "Requires optimizedTNT to be enabled";
```

Comment:
Why preventing it from setting first and then optimizedTNT.
Honestly I would break them down into two validators, cause in the rule description they could figure out as two entries. And one validator can cancel another out.

Reply:
Done